packages/material-ui/src/Popover/Popover.js

```diff
@@ -440,7 +440,12 @@ Popover.propTypes = {
    * It's used to set the position of the popover.
    */
   anchorEl: chainPropTypes(PropTypes.oneOfType([HTMLElementType, PropTypes.func]), (props) => {
-    if (props.open && (!props.anchorReference || props.anchorReference === 'anchorEl')) {
+    if (
+      props.open &&
+      (!props.anchorReference || props.anchorReference === 'anchorEl') &&
+      // can't know when it'll resolve
+      typeof props.anchorEl !== 'function'
```

Comment:
Position at (0,0).

Reply:
I'm not following how a .propType is supposed to prevent it? Did you consider the comment before this check?

Reply:
From what I understand:

- this change "allows" (as no longer raising a warning) `anchorEl` to return a value that the popover can't use for positioning when `anchorEl` is a function getter.
- when the popover can't use the `anchorEl` for positioning, it displays it at the position (0,0). This is true until a correct value is found. This can lead to jumps: #8040

> can't know when it'll resolve

Why do we want to support this case? I think that the only case where this would be acceptable is if 1. the Popover stays hidden until the value is resolved 2. the Popover gets systematically notified when the value resolves.